orangewidget/utils/messagewidget.py

```diff
@@ -541,6 +541,8 @@ def __update(self):
             fulltext = ""
         self.__fulltext = fulltext
         self.setToolTip(self.__styled(self.__defaultStyleSheet, fulltext))
+        self.setVisible(True)
```

Comment:
Restore the visible state as it was at entry (use `not isHidden()` to get the correct value to restore).

Reply:
Still, this seems wrong.

I think `self.layout().activate()` below is the confounding factor in the jittering update.
I would suggest:
1) removing that line
2) left align the icon widget so it preserves its absolute position within MessageWidget event when it's wider then necessary.

```diff
diff --git a/orangewidget/utils/messagewidget.py b/orangewidget/utils/messagewidget.py
index 4ea6f2cf8..7b3f624eb 100644
--- a/orangewidget/utils/messagewidget.py
+++ b/orangewidget/utils/messagewidget.py
@@ -356,7 +356,7 @@ class MessagesWidget(QWidget):
         self.setLayout(QHBoxLayout())
         self.layout().setContentsMargins(2, 1, 2, 1)
         self.layout().setSpacing(0)
-        self.layout().addWidget(self.__iconwidget)
+        self.layout().addWidget(self.__iconwidget, alignment=Qt.AlignLeft)
         self.layout().addSpacing(4)
         self.layout().addWidget(self.__textlabel)
         self.layout().addWidget(self.__popupicon)
@@ -552,7 +552,6 @@ class MessagesWidget(QWidget):
         else:
             self.__popuptext = fulltext
         self.__popupicon.setVisible(bool(self.__popuptext))
-        self.layout().activate()
 
     def mousePressEvent(self, event):
         if event.button() == Qt.LeftButton:
```

Reply:
What role does `self.layout().activate()` play here? Left-align solves the jittering with or without `.activate()`. I'll throw it out anyway, it's probably called implicitly at some point.

Reply:
Scratch that, I'm leaving it in, else this might happen: 
![Screenshot 2020-01-14 at 14 14 44](https://user-images.githubusercontent.com/24586651/72347351-dd784b80-36cf-11ea-819f-574301acef07.png)